src/notelistview.cpp

```diff
@@ -298,7 +298,6 @@ void NoteListView::mousePressEvent(QMouseEvent *e)
         if (!oldIndexes.contains(index)) {
             if (e->modifiers() == Qt::ControlModifier) {
                 setSelectionMode(QAbstractItemView::MultiSelection);
-                auto oldIndexes = selectionModel()->selectedIndexes();
                 if (oldIndexes.contains(index) && oldIndexes.size() > 1) {
```

Comment:
This `oldIndexes.contains(index)` check makes no sense, btw. If we get here, we already know that `!oldIndexes.contains(index)` was `true` after all. So either this code can be simplified or there is a bug.

Reply:
Seems like a bogus check to me as well. It was probably a copy-and-paste mistake from here (including the shadowing I removed):

https://github.com/nuttyartist/notes/blob/7027131e309b2d374b4047c6b0c712ff5ab4f5ec/src/notelistview.cpp#L337-L345

I'll remove this check on a separate commit. Thanks for noticing it!

Reply:
Yeah, somebody really liked to copy/paste code... there's so much duplication. :/

Reply:
Not only that, but it seems like that bogus `oldIndexes.contains(index)` check is what kept things working as they should, lol...

Check for yourself: Remove that check, and then use Ctrl+Click to select notes in the list view. You can no longer select more than two notes...

I removed most of those checks, and de-selecting notes while holding Ctrl still works here...

It'd be great if you could also confirm it before I commit:

```diff
diff --git a/src/notelistview.cpp b/src/notelistview.cpp
index 4ad8a96..9681ffd 100644
--- a/src/notelistview.cpp
+++ b/src/notelistview.cpp
@@ -298,12 +298,8 @@ void NoteListView::mousePressEvent(QMouseEvent *e)
         if (!oldIndexes.contains(index)) {
             if (e->modifiers() == Qt::ControlModifier) {
                 setSelectionMode(QAbstractItemView::MultiSelection);
-                if (oldIndexes.contains(index) && oldIndexes.size() > 1) {
-                    selectionModel()->select(index, QItemSelectionModel::Deselect);
-                } else {
-                    setCurrentIndex(index);
-                    selectionModel()->setCurrentIndex(index, QItemSelectionModel::SelectCurrent);
-                }
+                setCurrentIndex(index);
+                selectionModel()->setCurrentIndex(index, QItemSelectionModel::SelectCurrent);
                 auto selectedIndexes = selectionModel()->selectedIndexes();
                 emit notePressed(selectedIndexes);
             } else {
```

Reply:
> Check for yourself: Remove that check, and then use Ctrl+Click to select multiple notes in the list view. You can no longer select more than two notes...

Well, you can't just remove a check that is guaranteed to be `false`, because that changes the behavior of the code as if it were `true`. You indeed need to remove the conditional code as well, like you ended up doing. :-)